config-common/src/lib.rs

```diff
@@ -20,10 +24,18 @@ pub fn read_config<TConfig>(
 where
     TConfig: serde::de::DeserializeOwned,
 {
-    let config = std::fs::read(config_path)
-        .map_err(|err| ErrorKind::ReadConfig(Some(config_path.to_owned()), Box::new(err)))?;
-    let mut config: toml::Value = toml::from_slice(&config)
-        .map_err(|err| ErrorKind::ReadConfig(Some(config_path.to_owned()), Box::new(err)))?;
+    let mut config: toml::Value = match std::fs::read(config_path) {
+        Ok(contents) => toml::from_slice(&contents)
+            .map_err(|err| ErrorKind::ReadConfig(Some(config_path.to_owned()), Box::new(err)))?,
+
+        Err(err) if err.kind() == std::io::ErrorKind::NotFound => {
+            toml::Value::Table(Default::default())
+        }
+
+        Err(err) => {
+            return Err(ErrorKind::ReadConfig(Some(config_path.to_owned()), Box::new(err)).into())
+        }
```

Comment:
nit: not a _huge_ fan of the whitespace here, but whatever. After all, the braces do a good enough job padding the various match arms, so why add more?

Reply:
Well, *I* didn't add the braces, so... (Notice also the missing semi-colon on line 36.)

Fixed now.